File: CH7/startwithvuejs/src/services/FoodProductService.ts
import Vue from 'vue'
import { FoodProduct } from '../entities/FoodProduct'

export class FoodProductService {
  static getById (id: number) {
    return Vue.axios.get<FoodProduct>(`/foodproducts/${id}`)
  }

  static create (foodProduct: FoodProduct) {
    return Vue.axios.post<FoodProduct>(`/foodproducts`, foodProduct)
  }

  static update (foodProduct: FoodProduct) {
    return Vue.axios.put<FoodProduct>(`/foodproducts`, foodProduct)
  }

  static deActivate (foodProduct: FoodProduct) {
    return Vue.axios.delete<FoodProduct>(`/foodproducts/${foodProduct.id}`)
  }

  static getByFoodService (foodService: string, page:number, pageSize:number) {
    return Vue.axios.get<Array<FoodProduct>>(`/foodproducts?foodService=${foodService}&page=${page}&pageSize=${pageSize}`)
  }
}

File: CH8/StartWithJEE8/StartWithJEE8-ejb/src/main/java/com/packt/delivery/abstraction/service/foodproduct/FoodProductService.java

package com.packt.delivery.abstraction.service.foodproduct;

import com.packt.delivery.abstraction.entity.FoodProduct;
import java.util.List;
import java.util.Optional;

public interface FoodProductService {
    Optional<FoodProduct> getById(Integer id);
    FoodProduct save(FoodProduct foodProduct);
    FoodProduct update(FoodProduct foodProduct);
    FoodProduct deActivate(Integer id);
    List<FoodProduct> getByFoodService(String email, Integer page, Integer pageSize);    
}
